### PR TITLE
Update: option to run single test & disable coverage

### PR DESCRIPTION
**Pull request by sonhanguyen**
To run a specific test file:
`$ npm run test[:watch] --file=<absolute path or path relative to project root>`
watch mode has coverage off by default, to turn it on:
`$ npm run test:watch --coverage`

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+let env = '';
+const { f, file = f || argv[0] } = argv;
+if (file) env = `ADSLOT_TEST_FILE=${file} `;
+
+/* has to be done this way as karma removed the cli option to override config
+ https://stackoverflow.com/questions/29150998/karma-running-a-single-test-file-from-command-line
+ */
+
+ // npm run test [watch] -- [--file=] <absolute path>
```
Comment (sonhanguyen) [inline scripts/test]: not sure how to get rid of the first double dash

File: karma.conf.js [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+const webpack = require('webpack');
+const _ = require('lodash');
+
+const env = { ADSLOT_TEST_FILE: '' };
+
+webpackConfig.plugins.push(
+  new webpack.DefinePlugin(
+    _(env)
+      .assign(_.pick(process.env, _.keys(env)))
+      .mapValues(JSON.stringify)
```
Comment (sonhanguyen) [inline karma.conf.js]: the value will be inlined so it has to be defined like `{ ADSLOT_TEST_FILE: '"/some/path"' }`

File: scripts/test [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env node
+const TASK = 2;
+const argv = require('minimist')(process.argv.slice(TASK));
+console.log(argv)
```
Comment (vinteo) [inline scripts/test]: remove?

File: scripts/test [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env node
+const TASK = 2;
+const argv = require('minimist')(process.argv.slice(TASK));
```
Comment (vinteo) [inline scripts/test]: does this need to be added as a dev dependency?

Comment (sonhanguyen) [reply scripts/test]: it's already there

Comment (lightbringer1991) [reply scripts/test]: interesting reads
https://npmcompare.com/compare/commander,minimist,nomnom,optimist,yargs

Comment (sonhanguyen) [reply scripts/test]: there is also a lib called `vorpal` for interactive shell

File: scripts/test [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const path = require('path');
+const argv = require('minimist')(process.argv.slice(TASK));
+const { execSync } = require('child_process');
+
+let { file = argv.f || argv._.pop() } = argv;
+if (file) {
+  if (!path.isAbsolute(file)) {
+    file = execSync('npm root').toString().replace('node_modules', file).trim()
+  }
+  console.info('\x1b[4mRunning test:\x1b[0m\x1b[1m ', file, '\x1b[0m');
```
Comment (omgaz) [inline scripts/test]:
```
const colorGreen = '\x1b[4m';
console.info(`${colorGreen}Running test:\x1b[0m\x1b[1m ', file, '\x1b[0m`); 
```

**Comment by lightbringer1991** [comment]
looking great.
Maybe include an instruction on how to run a single test and/or no coverage

**Comment by sonhanguyen** [comment]
because of the way npm passes argument, it's annoying to use named argument like `npm run test -- --file=path`, which is why I made the `--file` part to be optional.

I settle with it but for reference, there is another approach which is #3fb38b7 which has a drawback of making the script dependent on npm to parse argument which means you can just run `node scripts/test --argument`.

**Comment by sonhanguyen** [comment]
@lightbringer1991 there is a separate branch for coverage off but I'm not sure it's really needed.

**Comment by lteacher** [comment]
I usually just use `only` with watch on, but lgtm :shirt:

**Comment by lightbringer1991** [comment]
@lteacher `.only` doesn't work for me, not sure why. That's why I proposed this change.

**Comment by omgaz** [comment]
LGTM

**Comment by sonhanguyen** [comment]
per feedback, added coverage flag

**Comment by omgaz** [comment]
LGTMerging 🍜

**Pull request merged**